addon/webextension/background/analytics.js

```diff
@@ -12,27 +12,30 @@ this.analytics = (function() {
     // sendTiming is only called in response to sendEvent, so no need to check
     // the telemetry pref again here.
     let timingCategory = "addon";
-    let url = main.getBackend() + "/timing";
-    let req = new XMLHttpRequest();
-    req.open("POST", url);
-    req.setRequestHeader("content-type", "application/json");
-    req.onload = catcher.watchFunction(() => {
-      if (req.status >= 300) {
-        let exc = new Error("Bad response from POST /timing");
-        exc.status = req.status;
-        exc.statusText = req.statusText;
-        console.error(exc);
-        // TODO: send the error to sentry, maybe?
-      }
+    return new Promise((resolve, reject) => {
+      let url = main.getBackend() + "/timing";
+      let req = new XMLHttpRequest();
+      req.open("POST", url);
+      req.setRequestHeader("content-type", "application/json");
+      req.onload = catcher.watchFunction(() => {
```

Comment:
Technically I don't think this will reject the promise, if there's an exception in the function then the promise will just hang. But it will report something to Sentry.